### ISSUE TITLE
Error when opening top-level or sample project in Android Studio

### ISSUE THREAD

**Issue by lgtout** [nhaarman/Acorn#116]
**Describe the bug**

I'm unable to successfully import either the top-level project or one of the sample projects into Android Studio or IDEA.  When I do so, the `Build` output console in the IDE displays an error.

**To Reproduce**
1. git clone the top-level project
2. In either IDE: Import the project.  In Android Studio - In the IDE welcome dialog select `Open an existing Android Studio project`.  Select the directory containing Acorn.  The IDE loads the project.
3. When the project is done loading the following error is displayed in the Build output console:
> Unable to load class 'org.gradle.execution.taskgraph.TaskInfo'.
Possible causes for this unexpected error include:<ul><li>Gradle's dependency cache may be corrupt (this sometimes occurs after a network connection timeout.)
Re-download dependencies and sync project (requires network)</li><li>The state of a Gradle build process (daemon) may be corrupt. Stopping all Gradle daemons may solve this problem.
Stop Gradle build processes (requires restart)</li><li>Your project may be using a third-party plugin which is not compatible with the other plugins in the project or the version of Gradle requested by the project.</li></ul>In the case of corrupt Gradle processes, you can also try closing the IDE and then killing all Java processes.

**Expected behavior**
I should be able to import the project and perform a build successfully.

**Stack trace**
Here's an abbreviated version of the stack trace:
![acorn____dev_acorn_](https://user-images.githubusercontent.com/364799/52181648-064a4e00-27a9-11e9-9749-43fcf2b9b39f.jpg)

**Environmental info (please complete the following information):**
 - Intellij IDEA 2018.3.4, Android Studio 3.3
 - OS: Mac Mojave 10.14.2

**Comment by nhaarman** [comment]
The sample projects are part of the entire projects, so it is intended to open the top-level project.

I'm not really sure what this error causes, but currently the master branch uses AGP `3.4.0-alpha10`. When I import it into AS 3.3 I get a specific error:

```
FAILURE: Build failed with an exception.

* What went wrong:
This version of Android Studio cannot open this project, please retry with Android Studio 3.4 or newer.
```

You could try downgrading the AGP to 3.3.0:

```diff
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- build.gradle	(revision a4853b07ee031dd07614f82121a271f19ebcfb95)
+++ build.gradle	(date 1549225717000)
@@ -8,7 +8,7 @@
 
     configurations.maybeCreate("pitest")
     dependencies {
-        classpath("com.android.tools.build:gradle:3.4.0-alpha10")
+        classpath("com.android.tools.build:gradle:3.3.0")
         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
         classpath("de.mannodermaus.gradle.plugins:android-junit5:1.3.2.0")
         classpath("com.adarshr:gradle-test-logger-plugin:1.6.0")
```

When I do this, the project syncs and builds fine in AS 3.3.

Does this work for you?

**Comment by lgtout** [comment]
Nevermind!  It's not you, it's ME.

After trying `./test` at the command line, I got this:
<pre>
> FAILURE: Build failed with an exception.

* Where:
Initialization script '/Users/bayo/.gradle/init.gradle' line: 10

* What went wrong:
Failed to apply plugin [class 'com.dorongold.gradle.tasktree.TaskTreePlugin']
> Could not create task ':taskTree'.
   > Could not generate a proxy class for class com.dorongold.gradle.tasktree.TaskTreeTask.
</pre>

Sure enough, the source of the problem was in the file specified above.  Removing `TaskTreePlugin` allowed me to run `./test`.  

Then I discovered that I couldn't import the project into Android Studio 3.3.  I got a message saying a more recent version of the IDE is required.  

That was solved by downloading and installing Android Studio 3.4 Preview.  But then AS gave a dialog saying `The project is using an incompatible version of the Android Gradle plugin`, with the option to have AS update the plugin.  I chose that and was able to complete the import of the project into AS.
